src/app/navigation/application-launcher.component.ts

```diff
@@ -0,0 +1,79 @@
+import {
+    Component, ElementRef, EventEmitter,
+    Input, OnInit, Output, Renderer2,
+    TemplateRef, ViewEncapsulation
+} from '@angular/core';
+
+import { NavigationItemConfig } from './navigation-item-config';
+import { read } from 'fs-extra';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'pfng-application-launcher',
+    templateUrl: './application-launcher.component.html'
+})
+
+export class ApplicationLauncherComponent  implements OnInit {
+
+    /**
+     *  Use a custom label for the launcher, default: Application Launcher
+     */
+    @Input() label: string;
+
+
+    /**
+     * Disable the application launcher button, default: false
+     */
+    @Input() isDisabled: boolean;
+
+
+    /**
+     * Display items as a list instead of a grid, default: false
+     */
+    @Input() isList: boolean;
+
+
+    /**
+     * Flag to not show icons on the launcher, default: false
+     */
+    @Input() hiddenIcons: boolean;
+
+
+    /**
+     * The navigation items used to build the menu
+     */
+    @Input() items: NavigationItemConfig[];
+
+
+    /**
+     * Internal boolean to toggle launcher, default:false
+     */
+    public isClicked: boolean;
+
+
+    /**
+     * The default constructor
+     */
+    constructor() {}
+
+    /**
+     * Initialize variable
+     */
+    ngOnInit(): void {
+      this.isClicked = false;
+    }
+
+    /**
+     * toggle function for launcher, active when click, return false on isDisabled:true
+     */
+   public toggle() {
```

Comment:
Can this be made private?

Reply:
@dlabrecq: I am binding to click event ```(click)="toggle()" ``` and keeping that private won't be allowed by the aot?

Reply:
I believe if the 'internal boolean' isClicked (open) property is made private, we can probably make the toggle method private as well. If the AOT compilation doesn't allow that, I'm fine with it as is.